lib/Controller/RoomController.php

```diff
@@ -659,6 +724,40 @@ public function addParticipantToRoom(string $newParticipant, string $source = 'u
 				return new DataResponse([]);
 			}
 
+			\call_user_func_array([$this->room, 'addUsers'], $participantsToAdd);
+		} else if ($source === 'circles') {
+			if (!$this->appManager->isEnabledForUser('circles')) {
+				return new DataResponse([], Http::STATUS_BAD_REQUEST);
+			}
+
+			/** @var Circles $circlesApi */
+			try {
+				$circle = Circles::detailsCircle($newParticipant);
+			} catch (\Exception $e) {
+				return new DataResponse([], Http::STATUS_NOT_FOUND);
+			}
+
+			$participants = [];
+			foreach ($circle->getMembers() as $member) {
+				/** @var Member $member */
+				if ($member->getType() === Member::TYPE_USER || $member->getUserId() === '') {
```

Comment:
I would add a check on the Level: $member->isLevel(MEMBER::LEVEL_MEMBER) as in current state, non-yet members (invited or requesting for access) will be invited to the room.

Reply:
Invites would be fine I guess. requests is bad of course :D

Reply:
in that case, have a look to ->getStatus() that should returns a string related to https://github.com/nextcloud/circles/blob/12934e8c5c886401231fb5808022891824250bc7/lib/Model/BaseMember.php#L41-L46 :]